src/incubator/TextField/Label.tsx

```diff
@@ -0,0 +1,43 @@
+import React, {useContext} from 'react';
+import {TextStyle} from 'react-native';
+import {Colors} from '../../style';
+import Text, {TextPropTypes} from '../../components/text';
+import {ColorType, ValidationMessagePosition} from './types';
+import {getColorByState} from './Presenter';
+import FieldContext from './FieldContext';
+
+export interface LabelProps {
+  label?: string;
+  labelColor?: ColorType;
```

Comment:
Why not try to deprecate the 'labelColor' and 'labelStyle' props and stay with 'labelProps' only? (since 'style' can be passed as a prop and 'color' as a style prop...)

Reply:
I always prefer to flatten and separate these two. 

In general, I believe specific props should be added for common cases and I think it's pretty common for the user to change the label style (without passing any other extra prop). 
Let's say it was very common to change the label letter spacings (it's clearly not), then I would consider adding a specific prop for it.